concent_api/core/migrations/0003_add_and_update_deadline_and_size.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.9 on 2018-06-19 09:37
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db import models
+from common.helpers import deserialize_message
+from common.helpers import parse_timestamp_to_utc_datetime
+
+
+def populate_subtask_size_field(apps, _schema_editor):
+    Subtask = apps.get_model('core', 'Subtask')
+    StoredMessage = apps.get_model('core', 'StoredMessage')
+    for subtask in Subtask.objects.filter(result_package_size__isnull=True):
+        if StoredMessage.objects.filter(
+                pk=subtask.report_computed_task_id).exists():
+            subtask.result_package_size = deserialize_message(
+                StoredMessage.objects.get(pk=subtask.report_computed_task_id).data).size
+            subtask.full_clean()
+            subtask.save()
+
+
+def populate_deadline_size_field(apps, _schema_editor):
+    Subtask = apps.get_model('core', 'Subtask')
+    StoredMessage = apps.get_model('core', 'StoredMessage')
+    for subtask in Subtask.objects.filter(computation_deadline__isnull=True):
+        if StoredMessage.objects.filter(
+                pk=subtask.task_to_compute_id).exists():
+            subtask.computation_deadline = parse_timestamp_to_utc_datetime(deserialize_message(
+                StoredMessage.objects.get(pk=subtask.task_to_compute_id).data).compute_task_def['deadline'])
+            subtask.full_clean()
+            subtask.save()
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ('core', '0002_subtask_force_get_task_result'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='subtask',
+            name='computation_deadline',
+            field=models.DateTimeField(blank=True, null=True),
+        ),
+        migrations.AddField(
+            model_name='subtask',
+            name='result_package_size',
+            field=models.IntegerField(blank=True, null=True),
+        ),
+        migrations.RunPython(
+            populate_subtask_size_field,
```

Comment:
You should use `RunPython.noop` as the reverse function, otherwise migration can't be reversed.

Reply:
Done.